lib/abilities.ex

```diff
@@ -9,5 +9,25 @@ defimpl Canada.Can, for: Pairmotron.User do
   def can?(%User{id: user_id}, action, %Group{owner_id: user_id})
     when action in [:edit, :update, :delete], do: true
 
+  def can?(%User{id: user_id}, :show, project = %Project{}) do
+    if is_nil(project.group_id) do
+      false
+    else
+      project = project |> Pairmotron.Repo.preload([{:group, :users}])
+      project.group.users
+      |> Enum.any?(fn user -> user.id == user_id end)
+    end
+  end
```

Comment:
So I added the edge case here, where users are allowed to look at a project that does not have a group so long as they have a retrospective that is associated with that project.

There is also an edge case to that, where they shouldn't be able to see a project that they have a retrospective associated with if that project is associated with a group that the user is no longer in. This is to prevent a user that has been removed from a group from seeing anything they aren't supposed to, as a project's resources should be private to members of that group.